### ISSUE TITLE
loss calculation

### ISSUE THREAD

**Issue by Lyt859165290** [fcdl94/MiB#12]
In the file loss.py, under class UnbiasedCrossEntropy(nn.Module) line 102
outputs[:, old_cl:] = inputs[:, old_cl:] - den.unsqueeze(dim=1) 
inputs[] is the raw logits of network, den is the logsumexp of those logits. To my understanding input[] should be first go through log and exp before operate with den.

**Comment by fcdl94** [comment]
Thank you for the issue.
However, I checked the loss and the calculation is right. 
We don't need to make the logsumexp of input[:, old_cl:] since we compute the probability of each input score, that means making logsumexp(x) where x is just one value, so that logsumexp(x).
The mathematics behind is that log((exp(input[i])) / sum(exp(input[:]))) can be computed as
input[i] - log(sum(exp(input[:]))).
